src/state_machine/coordinator/fire.rs

```diff
@@ -2977,9 +2921,13 @@ pub mod test {
         }
 
         // again remove signers so the number is insufficient
-        for _ in 0..num_signers_to_remove {
-            malicious.push(insufficient_signers.pop().unwrap());
-        }
+        malicious.extend(
+            insufficient_signers.drain(
+                insufficient_signers
+                    .len()
+                    .saturating_sub(num_signers_to_remove)..,
+            ),
+        );
```

Comment:
I find the above difficult to read, same thing on line 2870.
```suggestion
        let num_signers_to_drain = insufficient_signers
            .len()
            .saturating_sub(num_signers_to_remove);
        malicious.extend(insufficient_signers.drain(num_signers_to_drain..));
```

Reply:
Done in 6e184e0d59c1bab09584184e2a40244e1849f5ce